app/waf/src/controllers/adc.controller.ts

```diff
@@ -116,7 +110,7 @@ export class AdcController extends BaseController {
       tenantId: await this.reqCxt.get(WafBindingKeys.Request.KeyTenantId),
       reqId: this.reqCxt.name,
     };
-    this.adcStCtr = new AdcStateCtrlr(adc, addonReq);
+    this.adcStCtr = new AdcStateCtrlr(adc, addonReq, this.asgMgr);
```

Comment:
It's needless to pass the 3rd parameter to `AdcStateCtrlr`. 
If you are concerning the unit tests and want to stub it up with Sinon, why not stub it with:
`let stubInst = sinon.stub(AdcStateCtrlr.prototype, "some-function"); .. stubInst.callfake(..); ..`.
Have you searched online for it, or it doesn't work for you?